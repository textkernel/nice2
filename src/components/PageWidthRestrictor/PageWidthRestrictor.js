import React from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import styles from './PageWidthRestrictor.scss';

const { block } = bem({
    name: 'PageWidthRestrictor',
    classnames: styles
});

const PageWidthRestrictor = props => {
    const { children, As, ...rest } = props;

    return (
        <As {...rest} {...block(props)}>
            {children}
        </As>
    );
};

PageWidthRestrictor.displayName = 'PageWidthRestrictor';

PageWidthRestrictor.propTypes = {
    /** Node(s) to be rendered inside the container */
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
    /** HTML tag to be used to render the container */
    As: PropTypes.string
};

PageWidthRestrictor.defaultProps = {
    As: 'div'
};

export default PageWidthRestrictor;
