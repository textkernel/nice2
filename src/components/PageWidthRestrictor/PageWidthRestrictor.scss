@import '../../themes/oneui/oneui.scss';

.PageWidthRestrictor {
    min-width: var(--site-container-min-size);
    max-width: var(--site-container-max-size);
    padding: 0 var(--site-container-padding);
    margin: 0 auto;
}
