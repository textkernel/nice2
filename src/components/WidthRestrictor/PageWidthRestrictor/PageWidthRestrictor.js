import React from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import styles from './PageWidthRestrictor.scss';

const { block } = bem({
    name: 'PageWidthRestrictor',
    classnames: styles,
});

const PageWidthRestrictor = props => {
    const { children, ...rest } = props;

    return (
        <div {...rest} {...block(props)}>
            {children}
        </div>
    );
};

PageWidthRestrictor.displayName = 'PageWidthRestrictor';

PageWidthRestrictor.propTypes = {
    /** Node(s) to be rendered inside the container */
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
};

export default PageWidthRestrictor;
