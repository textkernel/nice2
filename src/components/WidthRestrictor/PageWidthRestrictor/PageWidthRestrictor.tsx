import * as React from 'react';
import { bem } from '../../../utils/bem';
import styles from './PageWidthRestrictor.scss';

interface Props extends React.HTMLAttributes<HTMLDivElement> {
    /** Node(s) to be rendered inside the container */
    children: NotEmptyReactNode;
}

const { block } = bem('PageWidthRestrictor', styles);

export const PageWidthRestrictor: React.FC<Props> = props => {
    const { children, ...rest } = props;

    return (
        <div {...rest} {...block(props)}>
            {children}
        </div>
    );
};

PageWidthRestrictor.displayName = 'PageWidthRestrictor';
