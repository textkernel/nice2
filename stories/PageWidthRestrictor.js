import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, withKnobs } from '@storybook/addon-knobs';
import { PageWidthRestrictor } from '@textkernel/oneui';

storiesOf('PageWidthRestrictor', module)
    .addDecorator(withKnobs)
    .add('PageWidthRestrictor', () => (
        <div style={{ backgroundColor: 'teal' }}>
            <PageWidthRestrictor
                As={text('Tag for component', 'div')}
                style={{ backgroundColor: 'darkturquoise' }}
            >
                {text('Content', 'This is a placeholder for children')}
            </PageWidthRestrictor>
        </div>
    ));
