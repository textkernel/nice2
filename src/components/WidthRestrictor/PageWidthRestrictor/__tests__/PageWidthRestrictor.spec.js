import React from 'react';
import toJson from 'enzyme-to-json';
import PageWidthRestrictor from '../PageWidthRestrictor';

describe('PageWidthRestrictor', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<PageWidthRestrictor>Some children</PageWidthRestrictor>);
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('div')).toHaveLength(1);
    });
});
