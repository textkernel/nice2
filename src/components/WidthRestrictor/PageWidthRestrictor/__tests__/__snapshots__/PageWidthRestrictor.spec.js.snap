// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageWidthRestrictor should render correctly 1`] = `
<div
  className="PageWidthRestrictor"
>
  Some children
</div>
`;
